src/internal/packager2/publish.go

```diff
@@ -0,0 +1,208 @@
+	if err != nil {
+		return fmt.Errorf("could not instantiate remote: %w", err)
+	}
+
+	// Execute copy
+	err = zoci.CopyPackage(ctx, srcRemote, dstRemote, opts.Concurrency)
+	if err != nil {
+		return fmt.Errorf("could not copy package: %w", err)
+	}
+
+	l.Debug("publisher2.PublishOCI done", "duration", time.Since(start))
+	return nil
+}
+
+// PublishPackageOpts declares the parameters to publish a package.
+type PublishPackageOpts struct {
+	// Concurrency configures the zoci push concurrency if empty defaults to 3.
+	Concurrency int
+	// SigningKeyPath points to a signing key on the local disk.
+	SigningKeyPath string
+	// SigningKeyPassword holds a password to use the key at SigningKeyPath.
+	SigningKeyPassword string
+	// SkipSignatureValidation flags whether Publish should skip validating the signature.
+	SkipSignatureValidation bool
+	// WithPlainHTTP falls back to plain HTTP for the registry calls instead of TLS.
+	WithPlainHTTP bool
+	// PublicKeyPath validates the create time signage of a package.
+	PublicKeyPath string
+	// Architecture is the architecture we are publishing to
+	Architecture string
+}
+
+// PublishPackage takes a Path to the location of the build package, a ref to a registry, and a PublishOpts and uploads the
+// package tarball, oci reference, or skeleton package to the registry.
+func PublishPackage(ctx context.Context, path string, dst registry.Reference, opts PublishPackageOpts) error {
+	l := logger.From(ctx)
+	// TODO: determine if the source is an OCI reference and a zoci.CopyPackage() is required
+	// TODO: can you copy to and from the same registry?
+
+	// Validate inputs
+	l.Debug("validating PublishOpts")
+	if err := dst.ValidateRegistry(); err != nil {
+		return fmt.Errorf("invalid registry: %w", err)
+	}
+	if path == "" {
+		return fmt.Errorf("path must be specified")
+	}
+
+	// Load package layout
+	l.Info("loading package", "path", path)
+	layoutOpts := layout2.PackageLayoutOptions{
+		PublicKeyPath:           opts.PublicKeyPath,
+		SkipSignatureValidation: opts.SkipSignatureValidation,
+	}
+	pkgLayout, err := layout2.LoadFromTar(ctx, path, layoutOpts)
+	if err != nil {
+		return fmt.Errorf("unable to load package: %w", err)
+	}
+
+	return pushToRemote(ctx, pkgLayout, dst, opts.Concurrency, opts.WithPlainHTTP)
+}
+
+// PublishSkeletonOpts declares the parameters to publish a skeleton package.
+type PublishSkeletonOpts struct {
+	// Concurrency configures the zoci push concurrency if empty defaults to 3.
+	Concurrency int
+	// SigningKeyPath points to a signing key on the local disk.
+	SigningKeyPath string
+	// SigningKeyPassword holds a password to use the key at SigningKeyPath.
+	SigningKeyPassword string
+	// WithPlainHTTP falls back to plain HTTP for the registry calls instead of TLS.
+	WithPlainHTTP bool
+}
+
+// PublishSkeleton takes a Path to the location of the build package, a ref to a registry, and a PublishOpts and uploads the
```

Comment:
Doccomment was copy-pasted from the original function when we refactored, needs to be trued up.